Implement QLDB Ledger resource and data lookup (#10394)
terraform-providers/terraform-provider-aws: website/docs/r/qldb_ledger.html.markdown

> ---

Reviewer: Since we're likely going to merge https://github.com/terraform-providers/terraform-provider-aws/pull/10684 shortly, we'll need to add a `subcategory` to this documentation page:

```suggestion
---
subcategory: "Quantum Ledger Database (QLDB)"
```

---
layout: "aws"
page_title: "AWS: aws_qldb_ledger"
description: |-
  Provides an QLDB Resource resource.
---

# Resource: aws_qldb_ledger

Provides an AWS Quantum Ledger Database (QLDB) resource

## Example Usage

```hcl
resource "aws_qldb_ledger" "sample-ledger" {
  name = "sample-ledger"
}
```

## Example with disabled deletion protection 

> **NOTE:** The deletion protection is enabled by default, so any changes that require a new resource will fail. 

Reviewer: I think this example can be omitted and the note wording is a little confusing. My suggestion would be to clarify the problem and solution inline:

```suggestion
~> **NOTE:** Deletion protection is enabled by default. To successfully delete this resource via Terraform, `deletion_protection = false` must be applied before attempting deletion. 
```

``` hcl
resource "aws_qldb_ledger" "sample-ledger" {
  name                = "sample-ledger"
  deletion_protection = false
}
```

## Argument Reference

The following arguments are supported:

* `name` - (Optional) The friendly name for the QLDB Ledger instance. By default generated by Terraform.
* `name_prefix` - (Optional) The friendly name for the QLDB Ledger instance. Conflicts with `name`.
* `permissions_mode` - (Optional) The permissions mode for the QLDB Ledger instance. More on [AWS documentation](https://docs.aws.amazon.com/qldb/latest/developerguide/ledger-management.basics.html)

> * `deletion_protection` - (Optional) The deletion protection for the QLDB Ledger instance. By default it is `true`.

Reviewer: ```suggestion
* `deletion_protection` - (Optional) The deletion protection for the QLDB Ledger instance. By default it is `true`. To delete this resource via Terraform, this value must be configured to `false` and applied first before attempting deletion.
```
